### ISSUE TITLE
Problems with UI

### ISSUE THREAD

**Issue by weter11** [SpecialKO/SpecialK#23]
Hi! There are some problems with GUI in my system. Launch:
![Снимок экрана_2022-08-23_11-43-07](https://user-images.githubusercontent.com/14630689/186114095-95868952-1baf-4412-9f5c-85283eaf896c.png)
Choose a game in a list from my steam library
![Снимок экрана_2022-08-23_11-43-31](https://user-images.githubusercontent.com/14630689/186114121-1b4e398f-03a3-4dda-8505-eb70f773c64c.png)

Used software: Ubuntu 22.04, Proton GE 7.30. I tried also Steam Proton 7.0, but result is the same, list of games is invisible and after launching a windows/linux game SKIF interface is absent. Log:
```
2022-08-23 11:11:48.336 INFO  [316] [SKIF_Initialize@3082] Special K Injection Frontend (SKIF) v 0.9.8
2022-08-23 11:11:48.337 INFO  [316] [SKIF_Initialize@3083] Built 21:31:21, May 30 2022
2022-08-23 11:11:48.337 INFO  [316] [SKIF_Initialize@3084] ----------------------------
2022-08-23 11:11:48.337 INFO  [316] [SKIF_Initialize@3085] Working directory:
2022-08-23 11:11:48.337 INFO  [316] [SKIF_Initialize@3086] Old: Z:\home\~\.local\share\Steam\steamapps\compatdata\3755445415\pfx\drive_c\users\steamuser\Documents\My Mods\SpecialK
2022-08-23 11:11:48.337 INFO  [316] [SKIF_Initialize@3087] New: Z:\home\~\.local\share\Steam\steamapps\compatdata\3755445415\pfx\drive_c\users\steamuser\Documents\My Mods\SpecialK
2022-08-23 11:11:48.739 INFO  [316] [SKIF_UI_Tab_DrawLibrary@918] Populating library list...
2022-08-23 11:11:48.739 INFO  [316] [SK_Steam_GetInstalledAppIDs@173] Detecting Steam games...
2022-08-23 11:11:48.741 INFO  [316] [SKIF_GOG_GetInstalledAppIDs@43] Detecting GOG games...
2022-08-23 11:11:48.741 INFO  [316] [SKIF_EGS_GetInstalledAppIDs@50] Detecting Epic games...
2022-08-23 11:11:48.741 WARN  [316] [SKIF_EGS_GetInstalledAppIDs@75] Failed to read Epic manifest location from the registry!
2022-08-23 11:11:48.741 INFO  [316] [SKIF_EGS_GetInstalledAppIDs@79] Epic manifest location: C:\ProgramData\Epic\EpicGamesLauncher\Data\Manifests\
2022-08-23 11:11:48.741 WARN  [316] [SKIF_EGS_GetInstalledAppIDs@84] Folder does not exist!
2022-08-23 11:11:48.741 INFO  [316] [SKIF_Xbox_GetInstalledAppIDs@28] Detecting Xbox games...
2022-08-23 11:11:48.741 INFO  [316] [SKIF_GetCustomAppIDs@220] Detecting custom SKIF games...
2022-08-23 11:11:48.741 INFO  [316] [SKIF_UI_Tab_DrawLibrary@969] Finished populating the library list.
2022-08-23 11:11:48.743 INFO  [424] [SKIF_UI_Tab_DrawLibrary::<lambda_5>::operator ()@976] Begin streaming game icons and names...
2022-08-23 11:11:48.770 INFO  [424] [SKIF_UI_Tab_DrawLibrary::<lambda_5>::operator ()@983] Loaded the embedded Patreon texture.
2022-08-23 11:11:48.782 INFO  [424] [SKIF_UI_Tab_DrawLibrary::<lambda_5>::operator ()@1169] Finished streaming game icons and names...
2022-08-23 11:16:35.060 INFO  [316] [wWinMain@6376] Writing last selected game to registry: 337000
```

**Comment by Aemony** [comment]
Hi,

Sadly we don't support Linux, and I didn't even expect SKIF or the injection service to be functional. In this case the issue would seem to be that while SKIF can read manifest files of... I assume it to be Steam... it cannot read the game titles of the Steam clients appinfo.vdf file (where localized titles etc are stored), as that would cause this particular issue.

One thing you could try with is using a newer version of Special K/the frontend app.
Latest experimental installer as of right now is https://sk-data.special-k.info/repository/SpecialK_22.8.18.exe

If that doesn't fix it, then I'm not really sure what would, other than using Windows, of course.

Br,
Aemony

**Comment by weter11** [comment]
So, version on website was outdated. Now I have a bit more logs and Monitor tab UI is working now. So it say that it don't inject anything, but log was changed.
```
2022-08-23 17:05:47.962 INFO  [312] [SKIF_InjectionContext::SKIF_InjectionContext@299] Watching 32-bit PID file: Z:\home\wer\.local\share\Steam\steamapps\compatdata\3755445415\pfx\drive_c\users\steamuser\Documents\My Mods\SpecialK\Servlet\SpecialK32.pid
2022-08-23 17:05:47.962 INFO  [312] [SKIF_InjectionContext::SKIF_InjectionContext@301] Watching 64-bit PID file: Z:\home\wer\.local\share\Steam\steamapps\compatdata\3755445415\pfx\drive_c\users\steamuser\Documents\My Mods\SpecialK\Servlet\SpecialK64.pid
2022-08-23 17:05:47.963 INFO  [312] [wWinMain@1826] Max severity to log was set to 4
2022-08-23 17:05:47.963 INFO  [312] [wWinMain@1848] Clearing out old installers...
2022-08-23 17:05:47.963 INFO  [312] [wWinMain@1917] ----------------------------
2022-08-23 17:05:47.963 INFO  [312] [wWinMain@1918] Current Registry State:
2022-08-23 17:05:47.963 INFO  [312] [wWinMain@1919] Special K user data:   
2022-08-23 17:05:47.963 INFO  [312] [wWinMain@1920] SKIF app registration: Z:\home\wer\.local\share\Steam\steamapps\compatdata\3755445415\pfx\drive_c\users\steamuser\Documents\My Mods\SpecialK\SKIF.exe
2022-08-23 17:05:47.963 INFO  [312] [SKIF_RegisterApp@913] App registration was successful: Z:\home\wer\.local\share\Steam\steamapps\compatdata\3755445415\pfx\drive_c\users\steamuser\Documents\My Mods\SpecialK\SKIF.exe
2022-08-23 17:05:47.963 INFO  [312] [SKIF_RegisterApp@921] Updated central Special K userdata location: Z:\home\wer\.local\share\Steam\steamapps\compatdata\3755445415\pfx\drive_c\users\steamuser\Documents\My Mods\SpecialK
2022-08-23 17:05:47.963 INFO  [312] [wWinMain@1922] ----------------------------
2022-08-23 17:05:48.291 INFO  [312] [SKIF_InjectionContext::_RefreshSKDLLVersions@636] SpecialK32.dll : 22.8.18
2022-08-23 17:05:48.291 INFO  [312] [SKIF_InjectionContext::_RefreshSKDLLVersions@637] SKIFsvc32.exe  : 1.0.3.1
2022-08-23 17:05:48.292 INFO  [312] [SKIF_InjectionContext::_RefreshSKDLLVersions@654] SpecialK64.dll : 22.8.18
2022-08-23 17:05:48.292 INFO  [312] [SKIF_InjectionContext::_RefreshSKDLLVersions@655] SKIFsvc64.exe  : 1.0.3.1
2022-08-23 17:05:48.340 INFO  [312] [SKIF_UI_Tab_DrawLibrary@813] Populating library list...
2022-08-23 17:05:48.340 INFO  [312] [SK_Steam_GetInstalledAppIDs@173] Detecting Steam games...
2022-08-23 17:05:48.343 INFO  [312] [SKIF_GOG_GetInstalledAppIDs@43] Detecting GOG games...
2022-08-23 17:05:48.343 INFO  [312] [SKIF_EGS_GetInstalledAppIDs@50] Detecting Epic games...
2022-08-23 17:05:48.343 WARN  [312] [SKIF_EGS_GetInstalledAppIDs@75] Failed to read Epic manifest location from the registry!
2022-08-23 17:05:48.343 INFO  [312] [SKIF_EGS_GetInstalledAppIDs@79] Epic manifest location: C:\ProgramData\Epic\EpicGamesLauncher\Data\Manifests\
2022-08-23 17:05:48.343 WARN  [312] [SKIF_EGS_GetInstalledAppIDs@84] Folder does not exist!
2022-08-23 17:05:48.343 INFO  [312] [SKIF_Xbox_GetInstalledAppIDs@28] Detecting Xbox games...
2022-08-23 17:05:48.344 INFO  [312] [SKIF_GetCustomAppIDs@229] Detecting custom SKIF games...
2022-08-23 17:05:48.344 INFO  [312] [SKIF_UI_Tab_DrawLibrary@864] Finished populating the library list.
2022-08-23 17:05:48.344 INFO  [416] [SKIF_CheckForUpdates::<lambda_1>::operator ()@1168] Update Channel: 
2022-08-23 17:05:48.344 ERROR [416] [SKIF_CheckForUpdates::<lambda_1>::operator ()@1197] Could not find the update channel in repository.json!
2022-08-23 17:05:48.344 ERROR [416] [SKIF_CheckForUpdates::<lambda_1>::operator ()@1211] Using fallback channel: Website
2022-08-23 17:05:48.345 INFO  [420] [SKIF_UI_Tab_DrawLibrary::<lambda_5>::operator ()@871] Thread started!
2022-08-23 17:05:48.345 INFO  [420] [SKIF_UI_Tab_DrawLibrary::<lambda_5>::operator ()@872] Streaming game icons and names asynchronously...
2022-08-23 17:05:48.364 INFO  [420] [SKIF_UI_Tab_DrawLibrary::<lambda_5>::operator ()@879] Loaded the embedded Patreon texture.
2022-08-23 17:05:48.377 INFO  [420] [SKIF_UI_Tab_DrawLibrary::<lambda_5>::operator ()@1065] Finished streaming game icons and names asynchronously...
2022-08-23 17:05:48.377 INFO  [420] [SKIF_UI_Tab_DrawLibrary::<lambda_5>::operator ()@1070] Thread stopped!
2022-08-23 17:05:48.408 INFO  [440] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1339] Thread started!
2022-08-23 17:05:48.408 INFO  [440] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1340] Streaming game cover asynchronously...
2022-08-23 17:05:48.421 INFO  [440] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1558] Finished streaming game cover asynchronously...
2022-08-23 17:05:48.421 INFO  [440] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1562] Thread stopped!
2022-08-23 17:06:29.082 INFO  [448] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1339] Thread started!
2022-08-23 17:06:29.082 INFO  [448] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1340] Streaming game cover asynchronously...
2022-08-23 17:06:29.084 INFO  [448] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1558] Finished streaming game cover asynchronously...
2022-08-23 17:06:29.084 INFO  [448] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1562] Thread stopped!
2022-08-23 17:06:33.303 INFO  [312] [SKIF_InjectionContext::_StartStopInject@146] Attempting to START the service...
2022-08-23 17:06:33.400 INFO  [312] [SKIF_InjectionContext::_StartStopInject@245] The operation was successful.
2022-08-23 17:07:57.962 INFO  [512] [SKIF_UI_Tab_DrawLibrary::<lambda_6>::operator ()@1339] Thread started!
```
Also app successfully start some games, also it show one nonsteam game and show it in a list normally. 
And it always show in library tab, that injection dll is 32bit, maybe that the reason why it not launched 64bit games. I tried CTRL+SHIFT+BACKSPACE in games, but SpecialK interface is absent in 32bit games also, steam overlay was turned off. The problem with appinfo.vdf I think in its encodings and used by SKIF font, that support it and my system don't have that font.
`Off-topic` Texmod successfully insert mods in old DX9 games like Remember me or Batman, but it need to manually choose game executables and app is very simple compared to SKIF.
Reason why I tried is that there is an article, reddit topic and YT video about successfully start injecting in Nier Automata FAR mod with ingame FAR UI settings. But that is only about 1 game, when SKIF was designed to inject in any app.

**Comment by Aemony** [comment]
Closing this issue for now.

As it stands:
- SKIF doesn't support Linux nor do we even expect the injection service that SKIF uses to work since it relies on OS-level CBT hooks of Windows.

- The fonts SKIF uses are included with Windows -- even those fonts used for non-Latin characters.

- All regular Latin characters/text in SKIF uses the same font, which based on the screenshots _are_ being loaded in this case, which suggests the issue is possibly something else (unless all game names were using non-Latin characters)

- SKIF had a recent bug with reading Steam's appinfo.vdf encoding due to Valve having changed the format. This was fixed with SpecialKO/SKIF@17d4461 and have been pushed to all update channels.

- This issue (if Linux were officially supported, which it isn't) technically belongs in [SpecialKO/SKIF/issues](https://github.com/SpecialKO/SKIF/issues)
